layer/training/runtime/model_trainer.py

```diff
@@ -167,6 +334,46 @@ def _train(
                     self.logger,
                 )
 
+                def _monitor_system_metrics(stop: bool) -> None:
+                    start_time = 0
+                    start_cpu_used = 0
+                    log_data_runner = LogDataRunner(
+                        client=self.client,
+                        train_id=self.train_context.train_id,
+                        logger=self.logger,
+                    )
+
+                    def _metrics_step_function(step: int) -> int:
+                        global step_value  # pylint: disable=invalid-name
+                        step += 1
+                        step_value += step
+                        return min(step, 15)
+
+                    if self.train_context.is_remote:
```

Comment:
Instead of introducing `is_remote`, you could do smt like `not isinstance(self.train_context, LocalTrainContext)`. Also, this `if` block likely needs a comment.

Reply:
Much better, thanks!